### PR TITLE
CCITCARBON-364 Allow iterate_method from cli

### PR DESCRIPTION
**Pull request by guyyaakov1**
Added Iterate Method option from cli

### CODE REVIEW

File: teflo/cli.py [hunk 1]
```diff
@@ -98,6 +104,10 @@ def show(ctx, scenario, list_labels, vars_data, show_graph):
         cbn.list_labels()
     elif show_graph:
         cbn.showgraph(ctx, scenario_graph)
+    elif iterate_method:
```
Comment (JunqiZhang0) [inline teflo/cli.py:107]: This part should be added as a part of the ```elif show_graph```

File: teflo/teflo.py [hunk 2]
```diff
@@ -343,6 +345,12 @@ def _validate_labels(self):
                                       "scenario descriptor file" % label)
         return
 
+    def _validate_iterate_method_option(self):
```
Comment (JunqiZhang0) [inline teflo/teflo.py:348]: This method is not being called by any function, why do we need this? If the purpose is for unit tests, it should go to test  part

**Review by JunqiZhang0 — CHANGES_REQUESTED** [review]

File: teflo/teflo.py (hunk 2 above)
Comment (rujutashinde) [reply teflo/teflo.py:348]: Guy and i spoke about this , he was supposed to make changes and submit the PR again. He had put this method only for a unit test.As per our last conversation he is to remove this and provide another unit test

**Review by rujutashinde — CHANGES_REQUESTED** [review]
please resubmit with the changes we discussed over the meeting

**Review by rujutashinde — COMMENTED** [review]
LGTM I will let @JunqiZhang0  approve it

**Review by JunqiZhang0 — APPROVED** [review]
LGTM

**Pull request merged**